crates/eips/src/eip4844/sidecar.rs

```diff
@@ -32,6 +36,107 @@ pub struct BlobTransactionSidecar {
     pub proofs: Vec<Bytes48>,
 }
 
+impl IntoIterator for BlobTransactionSidecar {
+    type Item = BlobTransactionSidecarItem;
+    type IntoIter = alloc::vec::IntoIter<BlobTransactionSidecarItem>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.blobs
+            .into_iter()
+            .zip(self.commitments)
+            .zip(self.proofs)
+            .map(|((blob, commitment), proof)| BlobTransactionSidecarItem {
+                index: 0,
+                blob,
+                kzg_commitment: commitment,
+                kzg_proof: proof,
+            })
+            .collect::<Vec<_>>()
+            .into_iter()
+    }
+}
+
+/// A single blob sidecar.
+#[derive(Clone, Debug, Default, PartialEq, Eq, Hash)]
+#[repr(C)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct BlobTransactionSidecarItem {
+    /// The index.
+    pub index: usize,
+    /// The index.
+    pub blob: Blob,
```

Comment:
serde is problematic here because this will overflow in debug mode, 

we can move 

https://github.com/alloy-rs/alloy/blob/21bc07ebbcb327486fc260cb1a8fb9d23f4bae53/crates/rpc-types-beacon/src/sidecar.rs#L105-L108

into this crate